clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1625,6 +1626,34 @@ public void testHeartbeatThreadClose() throws Exception {
             assertFalse("Heartbeat thread active after close", threads[i].getName().contains(groupId));
     }
 
+    @Test
+    public void testAsyncOffsetCommitAfterCoordinatorBackToService() {
```

Comment:
This test case passes without the fix. It doesn't look like it even goes through the auto-commit path.

Reply:
The name should mention the fact that this case covers auto-commit.